AVOutput/AVOutputTV.cpp

```diff
@@ -0,0 +1,5263 @@
+        {
+            LOGINFO("%s: Failed to parse argument\n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if (isPlatformSupport("DolbyVisionMode") != 0) returnResponse(false);
+
+        // Since it is dolby vision mode, to should get only for dolby vision format
+        format = "DV";
+
+        if (getParamIndex(source,pqmode,format,sourceIndex,pqIndex,formatIndex) == -1) 
+	{
+            LOGERR("%s: getParamIndex failed to get \n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        err = getLocalparam("DolbyVisionMode",formatIndex,pqIndex,sourceIndex,dolbyMode, PQ_PARAM_DOLBY_MODE);
+        if( err == 0 ) 
+	{
+            response["dolbyVisionMode"] = getDolbyModeStringFromEnum((tvDolbyMode_t)dolbyMode);
+            LOGINFO("Exit : DolbyVisionMode Value: %d \n", dolbyMode);
+            returnResponse(true);
+        }
+        else 
+	{
+            returnResponse(false);
+        }
+
+    }
+
+    uint32_t AVOutputTV::setDolbyVisionMode(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFO("Entry\n");
+
+        std::string value;
+        std::string pqmode;
+        std::string source;
+        std::string format;
+        tvError_t ret  = tvERROR_NONE;
+
+        value = parameters.HasLabel("dolbyVisionMode") ? parameters["dolbyVisionMode"].String() : "";
+        returnIfParamNotFound(parameters,"dolbyVisionMode");
+
+        if (parsingSetInputArgument(parameters, "DolbyVisionMode",source, pqmode, format) != 0)
+        {
+            LOGERR("%s: Failed to parse the input arguments \n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if (isPlatformSupport("DolbyVisionMode") != 0) returnResponse(false);
+
+        if (validateInputParameter("DolbyVisionMode",value) != 0) 
+	{
+            LOGERR("%s: Range validation failed for DolbyVisionMode\n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if( !isCapablityCheckPassed( pqmode, source, format, "DolbyVisionMode" )) 
+	{
+            LOGERR("%s: CapablityCheck failed for DolbyVisionMode\n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if( isSetRequired("Current",source,"DV") ) 
+	{
+            LOGINFO("Proceed with setDolbyVisionMode\n\n");
+            ret = SetTVDolbyVisionModeODM(value.c_str());
+        }
+
+        if(ret != tvERROR_NONE) 
+	{
+            LOGERR("Failed to set DolbyVisionMode\n\n");
+            returnResponse(false);
+        }
+        else 
+	{
+            int params[3]={0};
+            params[0]=GetDolbyModeIndex(value.c_str());
+            format = "DV";
+            int retval= UpdateAVoutputTVParam("set","DolbyVisionMode",pqmode,source,format,PQ_PARAM_DOLBY_MODE,params);
+            if(retval != 0 ) 
+	    {
+                LOGWARN("Failed to Save Dolbyvision mode\n");
+            }
+            LOGINFO("Exit : Dolbyvision successful to value: %s\n", value.c_str());
```

Comment:
at 2742 returing true even is failure case

Reply:
done